From b8aebbe352fbc44c21381658c6cff941646c78d8 Mon Sep 17 00:00:00 2001
From: Wangming Shao <shaowangming@h-partners.com>
Date: Tue, 8 Nov 2022 18:12:23 +0800
Subject: [PATCH] Fix the header file location error and adjust the function
 and structure version.
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b8aebbe352fbc44c21381658c6cff941646c78d8
Modified-by-SEL: No


driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5WGEF

-------------------------------------------------------------------

Fixed the header file location error.
Rectify the missing member and function name errors in the structure.

Signed-off-by: Wangming Shao <shaowangming@h-partners.com>
Reviewed-by: Yicong Yang <yangyicong@huawei.com>
Reviewed-by: Yang Jihong <yangjihong1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 tools/perf/arch/arm/util/auxtrace.c   | 6 +++---
 tools/perf/arch/arm/util/pmu.c        | 2 +-
 tools/perf/arch/arm64/util/hisi-ptt.c | 7 +++----
 3 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/tools/perf/arch/arm/util/auxtrace.c b/tools/perf/arch/arm/util/auxtrace.c
index e282fcea5e05..a4e4bb1a5711 100644
--- a/tools/perf/arch/arm/util/auxtrace.c
+++ b/tools/perf/arch/arm/util/auxtrace.c
@@ -10,10 +10,10 @@
 #include <linux/zalloc.h>
 #include <api/fs/fs.h>
 
-#include "../../util/auxtrace.h"
-#include "../../util/debug.h"
+#include "../../../util/auxtrace.h"
+#include "../../../util/debug.h"
 #include "../../util/evlist.h"
-#include "../../util/pmu.h"
+#include "../../../util/pmu.h"
 #include "cs-etm.h"
 #include "arm-spe.h"
 #include "hisi-ptt.h"
diff --git a/tools/perf/arch/arm/util/pmu.c b/tools/perf/arch/arm/util/pmu.c
index 2223083c6825..887c8addc491 100644
--- a/tools/perf/arch/arm/util/pmu.c
+++ b/tools/perf/arch/arm/util/pmu.c
@@ -11,7 +11,7 @@
 
 #include "arm-spe.h"
 #include "hisi-ptt.h"
-#include "../../util/pmu.h"
+#include "../../../util/pmu.h"
 
 struct perf_event_attr
 *perf_pmu__get_default_config(struct perf_pmu *pmu __maybe_unused)
diff --git a/tools/perf/arch/arm64/util/hisi-ptt.c b/tools/perf/arch/arm64/util/hisi-ptt.c
index ba97c8a562a0..110b2edf3e6b 100644
--- a/tools/perf/arch/arm64/util/hisi-ptt.c
+++ b/tools/perf/arch/arm64/util/hisi-ptt.c
@@ -113,7 +113,6 @@ static int hisi_ptt_recording_options(struct auxtrace_record *itr,
 			}
 			evsel->core.attr.freq = 0;
 			evsel->core.attr.sample_period = 1;
-			evsel->needs_auxtrace_mmap = true;
 			hisi_ptt_evsel = evsel;
 			opts->full_auxtrace = true;
 		}
@@ -126,16 +125,16 @@ static int hisi_ptt_recording_options(struct auxtrace_record *itr,
 	 * To obtain the auxtrace buffer file descriptor, the auxtrace event
 	 * must come first.
 	 */
-	evlist__to_front(evlist, hisi_ptt_evsel);
+	perf_evlist__to_front(evlist, hisi_ptt_evsel);
 	evsel__set_sample_bit(hisi_ptt_evsel, TIME);
 
 	/* Add dummy event to keep tracking */
-	err = parse_event(evlist, "dummy:u");
+	err = parse_events(evlist, "dummy:u", NULL);
 	if (err)
 		return err;
 
 	tracking_evsel = evlist__last(evlist);
-	evlist__set_tracking_event(evlist, tracking_evsel);
+	perf_evlist__set_tracking_event(evlist, tracking_evsel);
 
 	tracking_evsel->core.attr.freq = 0;
 	tracking_evsel->core.attr.sample_period = 1;
-- 
2.33.0

